adapters/httpapi_curl.c

```diff
@@ -113,20 +113,77 @@ HTTP_HANDLE HTTPAPI_CreateConnection(const char* hostName)
             httpHandleData->hostURL = malloc(hostURL_size);
+
             if (httpHandleData->hostURL == NULL)
             {
                 LogError("unable to malloc");
                 free(httpHandleData);
                 httpHandleData = NULL;
             }
+            else if ((strcpy_s(httpHandleData->hostURL, hostURL_size, "https://") != 0) ||
+                    (strcat_s(httpHandleData->hostURL, hostURL_size, hostName) != 0))
+            {
+                LogError("unable to set hostURL");
+                free(httpHandleData->hostURL);
+                free(httpHandleData);
+                httpHandleData = NULL;
+            }
+            else if ((httpHandleData->curl = curl_easy_init()) == NULL)
+            {
+                LogError("unable to init cURL structure");
+                free(httpHandleData->hostURL);
+                free(httpHandleData);
+                httpHandleData = NULL;
+            }
             else
             {
-                if ((strcpy_s(httpHandleData->hostURL, hostURL_size, "https://") == 0) &&
-                    (strcat_s(httpHandleData->hostURL, hostURL_size, hostName) == 0))
+                // Check correct TLS library has been configured for cURL.
+                const struct curl_tlssessioninfo *info = NULL;
+                CURLcode result = curl_easy_getinfo(httpHandleData->curl, CURLINFO_TLS_SSL_PTR, &info);
+
+                if (result != CURLE_OK || info == NULL)
+                {
+                    LogError("unable to get cURL backend SSL info");
+                    free(httpHandleData->hostURL);
+                    free(httpHandleData);
+                    httpHandleData = NULL;
+                }
+                else if ((info->backend != CURLSSLBACKEND_OPENSSL) && (info->backend != CURLSSLBACKEND_WOLFSSL) &&
+                         (info->backend != CURLSSLBACKEND_MBEDTLS))
+                {
+                    LogError("curl_sslbackend (%d) currently used by cURL is not supported by the C SDK. Please configure and compile cURL to use OpenSSL, wolfSSL, or mbedTLS.",
+                              info->backend);
+                    free(httpHandleData->hostURL);
+                    free(httpHandleData);
+                    httpHandleData = NULL;
+                }
+                else
                 {
-                    httpHandleData->curl = curl_easy_init();
-                    if (httpHandleData->curl == NULL)
+                    // Check correct TLS library has been configured for C SDK.
+                    int32_t SDKSSL = CURLSSLBACKEND_NONE;
+#ifdef USE_OPENSSL
+                    SDKSSL = CURLSSLBACKEND_OPENSSL;
+                    char* SDKSSLName = "OpenSSL";
+#elif USE_WOLFSSL
+                    SDKSSL = CURLSSLBACKEND_WOLFSSL;
+                    char* SDKSSLName = "wolfSSL";
+#elif USE_MBEDTLS
+                    SDKSSL = CURLSSLBACKEND_MBEDTLS;
+                    char* SDKSSLName = "mbedTLS";
```

Comment:
we should probably add `schannel` and `bearssl` to this list since those are options in curl as well and we allow those to be configured.

![image](https://user-images.githubusercontent.com/36710865/167177437-4c73b3ac-32dc-44fd-975b-da51445a04b1.png)

Reply:
If either of those are used by cURL, the LogError will be: 
LogError("curl_sslbackend (%d) currently used by cURL is not supported by the C SDK. Please configure and compile cURL to use OpenSSL, wolfSSL, or mbedTLS.", info->backend);

If either of those are used by CSDK, the LogError will be: 
LogError("C SDK TLS platform is not supported to use with cURL. Please configure and compile C SDK to use OpenSSL, wolfSSL, or mbedTLS.");

I don't think its required.  We could add more info to specify what is being used by the C SDK, but that's just more to maintain in the future.  I'd rather keep the logging specifics to only the one's we do support.

Reply:
Added BearSSL and tested.
Schannel won't be called by this file, since it is used on Windows.  Can't even compile with use_schannel on linux, since it is missing a windows.h file. CMake has win32 use a different file (httpapi_winhttp.c).